src/main/java/tecna/CommandScanner.java

```diff
@@ -0,0 +1,162 @@
+package tecna;
+
+import java.util.Scanner;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Scans and processes the user's input.
+ * An <code>inputIndex</code> is the index of the corresponding task if the command is <i>mark, unmark, or delete</i>.
+ * An <code>inputTask</code> is the task required to be added via <i>todo, deadline, or event</i> commands.
+ *
+ * @author DennieDan.
+ */
+public class CommandScanner {
+    private final Scanner SCANNER;
+    private String input;
+    private int inputIndex;
+    private Task inputTask;
+    private String keyword;
+
+    public CommandScanner() {
+        this.SCANNER = new Scanner(System.in);
+    }
+
+    public int getInputIndex() {
+        return this.inputIndex;
+    }
+
+    public Task getInputTask() {
+        return this.inputTask;
+    }
+
+    public String getKeyword() {
+        return this.keyword;
+    }
+
+    /**
+     * Returns the latest input scanned by the scanner
+     */
+    public String getInput() {
+        return this.input;
+    }
+
+    /**
+     * Gets the user's input and parses into the chatbot's valid type command.
+     * @return the corresponding command type in <code>CommandType enum</code>.
+     */
+    public CommandType getRequest() {
```

Comment:
the naming feels a bit vague as it is not really clear what Request means. Especially because the input from the user is referred to as 'input' in other places in your code. Maybe can change the method name to getUserRequest()  or something like that to make it more obvious what the purpose of this method is